tests/python/gaia-ui-tests/gaiatest/apps/contacts/regions/set_ice_contacts.py

```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from marionette_driver import expected, By, Wait
+
+from gaiatest import GaiaApps
+from gaiatest.apps.base import Base
+from gaiatest.apps.contacts.app import Contacts
+from gaiatest.form_controls.header import GaiaHeader
+
+
+class SetIceContacts(Base):
+
+    _toggle_locator = (By.CSS_SELECTOR, 'gaia-switch[name="ice-contact-1-enabled"]')
+    _select_ice_contact_1_locator = (By.ID, 'select-ice-contact-1')
+    _header_locator = (By.ID, 'ice-settings-header')
+    _ice_header_locator = (By.ID, 'ice-header')
+    _ice_contact = (By.CSS_SELECTOR, 'ol#ice-group')
+
+    def __init__(self, marionette):
+        Base.__init__(self, marionette)
+        GaiaApps(marionette).switch_to_displayed_app()
+        el = self.marionette.find_element(*self._header_locator)
+        Wait(self.marionette).until(lambda m: el.location['x'] == 0 and el.is_displayed())
+    
+    def tap_toggle_set_ice_contact(self):
+         self.marionette.find_element(*self._toggle_locator).tap()
+         return SetIceContacts(self.marionette)
+
+    def tap_select_ice_contact(self):
+        self.marionette.find_element(*self._select_ice_contact_1_locator).tap()
+        from gaiatest.apps.contacts.app import Contacts
+        return Contacts(self.marionette)
+
+    def tap_go_back(self):
+        GaiaHeader(self.marionette, self._header_locator).go_back()
+        from gaiatest.apps.contacts.regions.settings_form import SettingsForm
+        return SettingsForm(self.marionette)
+
+    @property
+    def is_ice_contact_shown(self):
+        Wait(self.marionette).until(expected.element_displayed(*self._ice_header_locator))
+        contact = self.marionette.find_element(*self._ice_contact)
+        Wait(self.marionette).until(expected.element_displayed(contact))
+        return self.marionette.find_element(*self.contact).is_displayed()
```

Comment:
I don't get this, the Wait() call always makes this method be true or it times out on the wait call.
I suggest to name this method something like `wait_for_ice_contact_shown` and call that then inside the test.